lib/pbench/test/unit/server/conftest.py

```diff
@@ -148,6 +149,55 @@ def db_session(server_config):
     Database.db_session.remove()
 
 
+@pytest.fixture
+def server_config_shared_db(pytestconfig, monkeypatch):
+    """
+    Mock a pbench-server.cfg configuration as defined above.
+    However, we replace sqllite in-memory db with on-disk database
+    for unit tests that needs db session access across multiple processes.
+
+    Args:
+        pytestconfig: pytest environmental configuration fixture
+        monkeypatch: testing environment patch fixture
+
+    Returns:
+        a PbenchServerConfig object the test case can use
+    """
+    cfg_file = pytestconfig.cache.get("_PBENCH_SERVER_CONFIG", None)
+    tmp_db = Path(pytestconfig.cache.get("TMP", None)) / "tmp.db"
+
+    with fileinput.FileInput(cfg_file, inplace=True, backup=".bak") as file:
+        for line in file:
+            print(
+                line.replace(
+                    "db_uri = sqlite:///:memory:", f"db_uri = sqlite:///{tmp_db}"
+                ),
+                end=" ",
+            )
+    monkeypatch.setenv("_PBENCH_SERVER_CONFIG", cfg_file)
```

Comment:
Wouldn't it be safer to make a copy and redefine the environment variable to that without changing the original?

Also consider using a `yield` fixture to allow you to reset everything automatically when the test case completes.